src/wrapper/simulation_objects.py

```diff
@@ -216,37 +324,154 @@ def add_subscriptions(self) -> int:
     @staticmethod
     def from_simulation(simulation_object: net.edge, updater) -> "Track":
         # see: https://sumo.dlr.de/pydoc/sumolib.net.edge.html
-        result = Track(simulation_object.getID())
+        result = Edge(simulation_object.getID())
         result.updater = updater
 
+        # pylint: disable=protected-access
+        result._from = simulation_object.getFromNode().getID()
+        result._to = simulation_object.getToNode().getID()
+
         return result
 
     def add_simulation_connections(self) -> None:
-        # we don't need references to other nodes
+        self._from = [
+            node for node in self.updater.nodes if node.identifier == self._from
+        ][0]
+        self._to = [node for node in self.updater.nodes if node.identifier == self._to][
+            0
+        ]
```

Comment:
The formatting here is kind of weird, but I think the linter wants it like that? :D

Reply:
Yup, this is formatted with black